packages/yew/src/backend/mod.rs

```diff
@@ -0,0 +1,87 @@
+//! When imported, this module represents the selected backend target.
+//! This may be selected via feature flag:
+//!     - (default) web_sys
+//!     - (not yet supported) static_render
+
+use cfg_if::cfg_if;
+
+/// A backend with which Yew communicates to render applications. This trait is implemented
+/// by all of the possible rendering backends.
+pub trait RenderingBackend {
+    /// A window on screen, analogous to the `window` global object in JavaScript.
+    type Window;
+
+    /// The document residing in a window. Analogous to the `document` global in Javascript.
+    type Document;
+
+    /// A nonspecific DOM element.
+    type Element;
+
+    /// A nonspecific DOM node.
+    type Node;
+
+    /// An event that notifies of editable content changes.
+    type InputEvent;
+
+    /// A DOM text node.
+    type TextNode;
+
+    /// A DOM element that represents a button, analogous to HTML's <button />
+    type ButtonElement;
+
+    /// A DOM element that represents a user input, analogous to HTML's <input />
+    type InputElement;
+
+    /// A DOM element representing a text area.
+    type TextAreaElement;
+
+    /// Returns the current window. This function will panic if there is no available window.
+    fn get_window() -> Self::Window;
+
+    /// Returns the current document.
+    fn get_document() -> Self::Document;
+
+    /// Returns the `origin` of the current window.
+    fn get_origin() -> Result<String, anyhow::Error>;
+
+    /// Returns the `host` for the current document. Useful for connecting to the server which serves the app.
+    fn get_host() -> Result<String, anyhow::Error>;
+}
+
+cfg_if! {
+    if #[cfg(feature = "static_render")] {
+        unimplemented!("SSR and SSG are not implemented yet");
+    } else {
+        mod web_sys;
```

Comment:
This should be renamed to something like `web_sys_backend` so it doesn't get confused with the `web_sys` crate. It hurts readability as is.